File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_56_B.c
#include <stdio.h>

int main() {
    int N, i, mn, mx;
    scanf("%d", &N);
    int A[N];
    for(i = 0; i < N; i++) {
        scanf("%d", &A[i]);
    }
    mn = N + 1;
    mx = -1;
    for(i = 0; i < N; i++) {
        if((i + 1) != A[i]) {
            mn = (mn < i) ? mn : i;
            mx = (mx > i) ? mx : i;
        }
    }
    if(mx == -1) {
        printf("0 0\n");
    } else {
        int temp, start = mn, end = mx;
        while(start < end) {
            temp = A[start]; 
            A[start] = A[end];
            A[end] = temp;
            start++;
            end--;
        }
        int sorted = 1;
        for(i = 1; i < N; i++) {
            if(A[i-1] > A[i]) {
                sorted = 0;
                break;
            }
        }
        if(sorted) {
            printf("%d %d\n", mn + 1, mx + 1);
        } else {
            printf("0 0\n");
        }
    }
    return 0;
} // End of Code

File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_56_B.go

package main

import (
	"fmt"
	"sort"
)

func main() {
	var N int
	fmt.Scan(&N)

	A := make([]int, N)
	for i := range A {
		fmt.Scan(&A[i])
	}

	mn, mx := N+1, -1
	for i := 0; i < N; i++ {
		if i+1 != A[i] {
			if i < mn {
				mn = i
			}
			if i > mx {
				mx = i
			}
		}
	}

	if mx == -1 {
		fmt.Println("0 0")
	} else {
		temp := append(A[:mn], A[mn:mx+1]...)
		sort.Slice(temp, func(i, j int) bool {
			return temp[i] > temp[j]
		})
		A = append(temp, A[mx+1:]...)

		B := make([]int, len(A))
		copy(B, A)
		sort.Ints(B)

		if equal(A, B) {
			fmt.Println(mn+1, mx+1)
		} else {
			fmt.Println("0 0")
		}
	}
}

func equal(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i, v := range a {
		if v != b[i] {
			return false
		}
	}
	return true
}
// End of Code

In the Go version of your code, I've introduced a helper function `equal` to compare two slices. The Go standard library doesn't provide a built-in way to do this.
